From 7c6dbeac3ecebd409bde142601314ad58f9b0fc4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Stefan=20L=C3=B6ffler?= <st.loeffler@gmail.com>
Date: Sat, 18 Feb 2023 22:45:02 +0100
Subject: [PATCH] Fix display of version date/time with Qt6

Qt6 is much stricter when parsing a datetime string, so tell git to
strictly adhere to ISO8601 rather than producing a more readable string
to put in GitRev.h
---
 CMake/Modules/getGitRevInfo.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMake/Modules/getGitRevInfo.cmake b/CMake/Modules/getGitRevInfo.cmake
index c07e0903b..ff2a9283c 100644
--- a/CMake/Modules/getGitRevInfo.cmake
+++ b/CMake/Modules/getGitRevInfo.cmake
@@ -21,7 +21,7 @@ endif()
 # Try to run git to obtain the last commit hash and date
 if (GIT_FOUND)
 	execute_process(COMMAND "${GIT_EXECUTABLE}" "--git-dir=.git" "show" "--no-patch" "--pretty=%h" RESULT_VARIABLE HASH_RESULT OUTPUT_VARIABLE NEW_HASH OUTPUT_STRIP_TRAILING_WHITESPACE)
-	execute_process(COMMAND "${GIT_EXECUTABLE}" "--git-dir=.git" "show" "--no-patch" "--pretty=%ci" RESULT_VARIABLE DATE_RESULT OUTPUT_VARIABLE NEW_DATE OUTPUT_STRIP_TRAILING_WHITESPACE)
+	execute_process(COMMAND "${GIT_EXECUTABLE}" "--git-dir=.git" "show" "--no-patch" "--pretty=%cI" RESULT_VARIABLE DATE_RESULT OUTPUT_VARIABLE NEW_DATE OUTPUT_STRIP_TRAILING_WHITESPACE)
 
 	if (${HASH_RESULT} EQUAL 0 AND ${DATE_RESULT} EQUAL 0 AND NOT "${NEW_HASH}" STREQUAL "" AND NOT "${NEW_DATE}" STREQUAL "")
 		set(SUCCESS TRUE)
